src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -42,20 +42,48 @@ public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTyp
             IsParsable = TryGetParsability(parameter, wellKnownTypes, out var parsingBlockEmitter);
             ParsingBlockEmitter = parsingBlockEmitter;
         }
+        else if (parameter.HasAttributeImplementingInterface(wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromFormMetadata), out _))
+        {
+            Source = EndpointParameterSource.Unknown;
+        }
         else if (TryGetExplicitFromJsonBody(parameter, wellKnownTypes, out var isOptional))
         {
-            Source = EndpointParameterSource.JsonBody;
+            if (SymbolEqualityComparer.Default.Equals(parameter.Type, wellKnownTypes.Get(WellKnownType.System_IO_Stream)))
+            {
+                Source = EndpointParameterSource.SpecialType;
+                AssigningCode = "httpContext.Request.Body";
+
+            }
+            else if (SymbolEqualityComparer.Default.Equals(parameter.Type, wellKnownTypes.Get(WellKnownType.System_IO_Pipelines_PipeReader)))
+            {
+                Source = EndpointParameterSource.SpecialType;
+                AssigningCode = "httpContext.Request.BodyReader";
+            }
+            else
+            {
+                Source = EndpointParameterSource.JsonBody;
+            }
             IsOptional = isOptional;
         }
         else if (parameter.HasAttributeImplementingInterface(wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromServiceMetadata)))
         {
             Source = EndpointParameterSource.Service;
         }
+        else if (parameter.HasAttribute(wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_AsParametersAttribute)))
+        {
+            Source = EndpointParameterSource.Unknown;
+        }
         else if (TryGetSpecialTypeAssigningCode(Type, wellKnownTypes, out var specialTypeAssigningCode))
         {
             Source = EndpointParameterSource.SpecialType;
             AssigningCode = specialTypeAssigningCode;
         }
+        else if (SymbolEqualityComparer.Default.Equals(parameter.Type, wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_IFormFile)) ||
```

Comment:
Is `HttpContext` parameter supported?

Reply:
Yes, test coverage is [here](https://github.com/dotnet/aspnetcore/blob/9c7db555f5c8b7069347b9c62976bd951afba4a0/src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.cs#L23).